common/static/common/js/components/views/list.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Generic view to render a collection.
+ */
+;(function (define) {
+    'use strict';
+    define(['backbone', 'underscore'], function(Backbone, _) {
+        var ListView = Backbone.View.extend({
+            /**
+             * Override with the view used to render models in the collection.
+             */
+            itemViewClass: Backbone.View,
+
+            initialize: function (options) {
+                this.itemViewClass = options.itemViewClass || this.itemViewClass;
+                this.collection.on('add', this.render, this);
+                this.collection.on('remove', this.render, this);
+                this.collection.on('reset', this.render, this);
+                this.collection.on('sync', this.render, this);
+                this.collection.on('sort', this.render, this);
+                // Keep track of our children for garbage collection
+                this.itemViews = [];
+            },
+
+            /**
+             * Naive render.  Any time the collection changes at all we
+             * re-render the entire list due to ordering
```

Comment:
I probably would put this comment somewhere else. I believe that ``render`` is always assumed to be a complete rendering, and that if you want incremental re-rendering then you have to manually handle the child views. Is there a way for ``render`` to be able to know what changed since the last time somehow?